design/nfs/nfs-provisioner-controlled-by-operator.md

```diff
@@ -0,0 +1,82 @@
+---
+title: nfs-provisioner-controlled-by-operator
+target-version: release-X.X
+---
+
+# NFS Provisioner Controlled by Operator
+
+## Summary
+
+Rook NFS Provisioner is a built in dynamic provisioner for Rook NFS. The functionality works fine but has a bug which the provisioner uses the same directory between the provisioned PV.
+
+Step to reproduce can be found in this issue [#4982](https://github.com/rook/rook/issues/4982)
+
+### Goals
+
+- Make NFS Provisioner to create different directory for each PV.
+- Each NFS Server has own provisioner.
+- Improve reliability of NFS Provisioner.
+
+### Non-Goals
+
+- NFS Operator manipulates uncontrolled resources.
+
+## Proposal details
+
+The approach will be similar to [Kubernetes NFS Client Provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/nfs-client) which the provisioner mounts the whole of NFS share into the provisioner pod (by kubelet) then the provisioner creates the appropriate directory. Currently Rook NFS Provisioner deployed independently and before NFS Server itself we cannot mount the NFS share because we don't know what NFS Server IP and the export directory.
```

Comment:
Why do we mount nfs share on the provisioner pod?

Reply:
So that the provisioner can create the directory for each NFS PV.

Reply:
Okay

Reply:
Maybe part of my confusion is that *perhaps* we are using language and descriptions that conflates the PVs that back the NFS server shares, and the PVs that a user gets from making a PVC that points (via a storageClass) to a NFS share?